qiskit/transpiler/passes/routing/stochastic_swap.py

```diff
@@ -176,14 +179,21 @@ def _layer_permutation(self, layer_partition, layout, qubit_subset,
         # Scaling matrix
         scale = np.zeros((num_qubits, num_qubits))
 
-        int_qubit_subset = _regtuple_to_numeric(qubit_subset, qregs)
-        int_gates = _gates_to_idx(gates, qregs)
-        int_layout = nlayout_from_layout(layout, qregs, coupling.size())
+        int_qubit_subset = np.fromiter((self._qubit_indices[bit]
+                                        for bit in qubit_subset),
+                                       dtype=np.int32,
```

Comment:
Looking at these I do wonder we don't make these unsigned since we can't have negative qubit indices. But that would require more cython changes because this array gets passed as an arg to the `swap_trial` cython function.

Reply:
That's a good idea, maybe an issue as a future improvement (or as a target of opportunity next time we need to dive into the cython code)?

Reply:
yeah I can push a follow up to do this, it's not crtiical